### ISSUE TITLE
BUG in Mat::copyTo()

### ISSUE THREAD

**Issue by ereator** [opencv/opencv#7160]
##### System information (version)
- OpenCV => 2.4.13
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015

##### Detailed description

In the file copy.cpp, function void Mat::copyTo( OutputArray _dst ) const:
For the large matrices, getContinuousSize() function may produce values that exceed the maximum possible values for the Size sz variable. As a concequence, size_t len becomes unbounded.
This mistake leads to an exception.
In the OpenCV 3.1.0 this issue do not pop up. Please fix it for 2.4.13.

##### Steps to reproduce
```.cpp
	Mat acc;
	Mat X(1000000, 625, CV_8UC1);

	while (true) {
		if (acc.empty()) X.copyTo(acc);
		else acc.push_back(X);
		printf("%ld\n", acc.rows);
	}
```
The exception comes after "4000000".

**Comment by savuor** [comment]
This bug was already [mentioned](http://code.opencv.org/issues/3232) and solved at PR [#3987](https://github.com/opencv/opencv/pull/3987) in 3.0.
Need to port it to 2.4.

**Comment by savuor** [comment]
Resolved in #7218
